resolvers/product/index.js

```diff
@@ -0,0 +1,55 @@
+const data = require("./productListing.json");
+
+const ProductListingSort = {
+    A_Z: "A_Z",
+    Z_A: "Z_A",
+    LOW_TO_HIGH: "LOW_TO_HIGH",
+    HIGH_TO_LOW: "HIGH_TO_LOW",
+    POPULAR: "POPULAR",
+    NEW: "NEW",
+    FAVOURITES: "FAVOURITES"
+};
+
+function product(_ , { id }) {
+    return data.products.find((product) => product.id === id) || null;
+}
+
+function productSummaryListing(_, { sort }) {
+  let products = data.products;
+
+  switch (sort) {
+    case ProductListingSort.A_Z:
+      products = products.sort((a, b) => (a.name > b.name ? 1 : -1));
+      break;
+    case ProductListingSort.Z_A:
+      products = products.sort((a, b) => (a.name < b.name ? 1 : -1));
+      break;
+    case ProductListingSort.LOW_TO_HIGH:
+      products = products.sort((a, b) =>
+        a.defaultVariant.price.amount.amount >
+        b.defaultVariant.price.amount.amount
+          ? 1
+          : -1
+      );
+      break;
+    case ProductListingSort.HIGH_TO_LOW:
+      products = products.sort((a, b) =>
+        a.defaultVariant.price.amount.amount <
+        b.defaultVariant.price.amount.amount
+          ? 1
+          : -1
+      );
+      break;
+  }
+
+  let response = Object.assign({}, data, { products });
+
+  return response;
+}
+
+module.exports = {
+  product,
+  productListing: productSummaryListing,
+  productSummaryListing
```

Comment:
I am not sure I understand these 2 lines, cannot we use only `productListing` Query?

Reply:
I think `productSummaryListing` is the one we need, as it provides more options, like sorting, filtering, or we can modify the `productListing` as FE uses that query. We don't need both for sure, we just haven't considered simplifying it yet.

Reply:
Since the FE uses the `productListing`, it makes sense to keep the standard. I'll remove the `productSummaryListing`.